code/datums/mind.dm

```diff
@@ -1063,6 +1082,57 @@
 				log_admin("[key_name(usr)] has automatically forged objectives for [key_name(current)]")
 				message_admins("[key_name_admin(usr)] has automatically forged objectives for [key_name_admin(current)]")
 
+	else if(href_list["ninja"])
+		var/mob/living/carbon/human/H = current
+		switch(href_list["ninja"])
+			if("clear")
+				if(src in ticker.mode.ninjas)
+					ticker.mode.ninjas -= src
+					special_role = null
+					ticker.mode.update_ninja_icons_removed(src)
+					message_admins("[key_name_admin(usr)] has de-ninja'ed [current].")
+					log_admin("[key_name(usr)] has de-ninja'ed [current].")
+			if("equip")
+				qdel(H.w_uniform)
+				qdel(H.wear_suit)
+				qdel(H.glasses)
+				qdel(H.wear_mask)
+				qdel(H.head)
+				qdel(H.l_ear)
+				qdel(H.shoes)
+				qdel(H.gloves)
+				qdel(H.back)
+				qdel(H.belt)
+				H.equipOutfit(/datum/outfit/ninja)
+			if("nanotrasen")
+				if(!(src in ticker.mode.ninjas))
+					ticker.mode.ninjas += src
+					special_role = SPECIAL_ROLE_SPACE_NINJA
+					ticker.mode.update_ninja_icons_added(src)
+					generate_ninja_objectives(src, TRUE, TRUE)
+					greet_ninja(H, TRUE)
+					message_admins("[key_name_admin(usr)] has friendly ninja'ed [current].")
+					log_admin("[key_name(usr)] has friendly ninja'ed [current].")

```

Comment:
`current` is a mob, so please use `key_name(current)` in all of these logging procs.